esy-lib/RunAsync.ml

```diff
@@ -30,6 +30,14 @@ let both a b =
     | Error err, Error _ -> Error err
   )
 
+let ofResult ?err r = 
+    match r with
+    | Ok v -> return v
+    | Error _ ->
+            match err with
+            | Some e -> error e
+            | None -> error "Unknown error"
```

Comment:
why not use error from `Error` above here?

Reply:
Hmm, you mean like:
```
    | Error msg ->
            begin match err with
            | Some e -> error e
            | None -> error msg 
            end
```
?

I had trouble getting this to work with the type system:
```
Error: The implementation esy-lib/RunAsync.pp.ml
       does not match the interface esy-lib/.EsyLib.objs\esyLib__RunAsync.cmi:
       Values do not match:
         val ofResult : ?err:string -> ('a, string) result -> 'a Run.t Lwt.t
       is not included in
         val ofResult : ?err:string -> ('a, 'b) result -> 'a t
       File "esy-lib/RunAsync.mli", line 54, characters 0-61:
         Expected declaration
       File "esy-lib/RunAsync.ml", line 33, characters 4-12:
         Actual declaration
```

Let me know if you have any ideas, I'm probably missing something obvious

Reply:
I've added a `RunAsync.ofBosError` instead which lifts from `('v, [> | `Msg of string | `CommandError of ... ]) result` to `RunAsync.t`.